sql: fix dependent object type in rename error message

spilchen opened the pull request:
Previously, attempting to rename a table that is referenced by a trigger would result in an error message mislabeling the dependent object as a view, regardless of its actual type. For example:
```
  cannot rename relation "t" because view "xy" depends on it
```

This commit corrects the error message to accurately reflect the type of the dependent object, such as "table" in the case of a trigger.

Informs #142370

Epic: none
Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/146646)
<!-- Reviewable:end -->

fqazi reviewed (approved):
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/> 

Just one minor issue with this change

> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/146646#-:-OQ9p8Qh9Q1h6r-SHrQD:b-u7m6gr)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @kev-cao)

___
*[`pkg/sql/rename_table.go` line 298 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/146646#-OQ9p54H3X2r3HC9bHtG:-OQ9p54H3X2r3HC9bHtH:b1kzmas) ([raw file](https://github.com/cockroachdb/cockroach/blob/c43be2423e73b3be6ce7893307bc3478d84eed4d/pkg/sql/rename_table.go#L298)):*
> ```Go
> 	var dependentType string
> 	switch {
> 	case desc.IsTable():
> ```

desc.GetObjectTypeString() will do this for you


<!-- Sent from Reviewable.io -->

spilchen reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/146646)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @fqazi and @kev-cao)

___
*[`pkg/sql/rename_table.go` line 298 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/146646#-OQ9p54H3X2r3HC9bHtG:-OQ9sgzUCRdquVLd3EHQ:b95z0eo) ([raw file](https://github.com/cockroachdb/cockroach/blob/c43be2423e73b3be6ce7893307bc3478d84eed4d/pkg/sql/rename_table.go#L298)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

desc.GetObjectTypeString() will do this for you
</blockquote></details>

Ah, I knew there was a function like this. But for the life of me I couldn't find it. Thanks, I applied your suggestion.


<!-- Sent from Reviewable.io -->

rafiss commented on pkg/sql/rename_table.go:
```diff
@@ -286,25 +286,25 @@ func (p *planner) dependentFunctionError(
 		viewName = viewFQName.FQString()
 	}
-	return sqlerrors.NewDependentBlocksOpError(op, typeName, objName, "view", viewName)
+	return sqlerrors.NewDependentBlocksOpError(op, typeName, objName, desc.GetObjectTypeString(), viewName)
```
super nit: we can improve the `NewDependentBlocksOpError` function by making the `dependentType` parameter be of type `redact.SafeString`

spilchen replied:
Good suggestion. I believe `op` and `typeName` don't need to be redacted either. I opted to cast all three to `redact.SafeString` inside `NewDependentBlocksOpError` when formatting the message.

rafiss commented on pkg/sql/rename_table.go:
```diff
@@ -286,25 +286,25 @@ func (p *planner) dependentFunctionError(
 			return sqlerrors.NewDependentObjectErrorf(
-				"cannot %s %s %q because a view depends on it",
-				op, typeName, objName)
+				"cannot %s %s %q because a %q depends on it",
```
super nit: i think we should use `because a %s depends on it` (instead of `%q`). we don't need to quote the type of the object. also, we can cast `op` and `desc.GetObjectTypeString()` to `redact.SafeString` to avoid unnecessary redaction.

spilchen committed:
sql: fix dependent object type in rename error message

Previously, attempting to rename a table that is referenced by a trigger
would result in an error message mislabeling the dependent object as a
view, regardless of its actual type. For example:
```
  cannot rename relation "t" because view "xy" depends on it
```

This commit corrects the error message to accurately reflect the type
of the dependent object, such as "table" in the case of a trigger.

Informs #142370

Epic: none
Release note: none
